enginetest/queries/index_queries.go

```diff
@@ -4338,4 +4338,63 @@ var IndexQueries = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "indexes and if exists",
+		SetUpScript: []string{
+			"create table t (i int, j int);",
+			"create index idx on t (i);",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:       "create index idx on t(j)",
+				ExpectedErr: sql.ErrDuplicateKey,
+			},
+			{
+				Query: "create index if not exists idx on t(j)",
```

Comment:
I'm not sure MySQL actually supports this syntax, based on their docs: 
https://dev.mysql.com/doc/refman/9.3/en/create-index.html

I tested with MySQL 8.4, but didn't try with MySQL 9.x. 

That said, I think it's still useful and I'm not opposed to supporting it, although it will likely be hard for customers to discover if it isn't in the MySQL docs. We should just figure out if we want to be consistent with `CREATE TABLE IF NOT EXISTS` or `CREATE IF NOT EXISTS VIEW`, which unfortunately seems to be inconsistent in MySQL.

Reply:
Yeah, I noticed this too.
MariaDB has been doing `create view IF NOT EXISTS`, which _is_ consistent with all the other `CREATE ... IF NOT EXISTS` queries... it might be possible to do both?
So `CREATE [IF NOT EXISTS] VIEW [IF NOT EXISTS] ...`?

Reply:
I downloaded MySQL 9.3 and tested it out; their docs are just messed up.
```
mysql> select version();
+-----------+
| version() |
+-----------+
| 9.3.0     |
+-----------+
1 row in set (0.0003 sec)

mysql> create if not exists view v as select 1;
ERROR: 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'if not exists view v as select 1' at line 1

mysql> create view if not exists v as select 1;
Query OK, 0 rows affected (0.0044 sec)

mysql> select * from v;
+---+
| 1 |
+---+
| 1 |
+---+
1 row in set (0.0009 sec)
```